Compile error on R 3.3 with dev version

Like #225, when we built `arrow` with the master branch of `cpp11`, there is a compile error on R 3.3:

```
../inst/include/cpp11/r_vector.hpp: In function ‘SEXPREC* cpp11::writable::truncate(SEXP, R_xlen_t, R_xlen_t)’:
../inst/include/cpp11/r_vector.hpp:889:30: error: ‘length_’ was not declared in this scope
   x = safe[Rf_lengthgets](x, length_);
                              ^~~~~~~
../inst/include/cpp11/r_vector.hpp:889:30: note: suggested alternative: ‘length’
   x = safe[Rf_lengthgets](x, length_);
                              ^~~~~~~
                              length
```

See https://github.com/ursacomputing/crossbow/runs/3302757244?check_suite_focus=true#step:8:1992

It looks like a typo perhaps: https://github.com/r-lib/cpp11/compare/v0.3.1...master#diff-65ceb5633686ca7a39f13ea1af039ba5b7bbc7bd225ca4eedbe181cf097ab0e6R887

--- a/inst/include/cpp11/r_vector.hpp
+++ b/inst/include/cpp11/r_vector.hpp
@@ -884,7 +884,7 @@ inline SEXP truncate(SEXP x, R_xlen_t length, R_xlen_t capacity) {
   SET_TRUELENGTH(x, capacity);
   SET_GROWABLE_BIT(x);
 #else
-  x = safe[Rf_lengthgets](x, length_);
+  x = safe[Rf_lengthgets](x, length);
 #endif
   return x;
 }

